src/template.rs

```diff
@@ -46,9 +47,27 @@ pub fn create_liquid_object(
     liquid_object.insert("username".into(), Value::Scalar(authors.username.into()));
     liquid_object.insert("os-arch".into(), Value::Scalar(os_arch.into()));
 
+    liquid_object.insert(
+        "within_cargo_project".into(),
+        Value::Scalar(is_within_cargo_project(project_dir).into()),
+    );
+
     Ok(liquid_object)
 }
 
+fn is_within_cargo_project(project_dir: &Path) -> bool {
+    let mut cwd = PathBuf::from(project_dir);
+    cwd.push("dummy");
```

Comment:
is this needed?

Reply:
yeah, `path::with_file_name()` replaces the filename in the path, so if there's no filename to begin with (as `cwd` is a path to a folder), you end up with the wrong result.

alternatively:
```rust
fn is_within_cargo_project(project_dir: &Path) -> bool {
    let mut cwd = PathBuf::from(project_dir);
    cwd.push("Cargo.toml");
    if cwd.exists() {
        return true;
    }
    while cwd.pop() && cwd.exists() {
        if cwd.with_file_name("Cargo.toml").exists() {
            return true;
        }
    }
    false
}
```

Reply:
Found solution... coming up